### PR TITLE
Change the ASN1 variant of x942kdf so that it can test ACVP data.

### PR DESCRIPTION
**Pull request by slontis**
This 'special' way of specifying the data should only be used for testing
purposes. It should not be used in production environments.
ACVP passes a blob of DER encoded data for some of the fields rather
than passing them as separate fields that need to be DER encoded.

##### Checklist

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

**Review by paulidale — APPROVED** [review]

File: providers/implementations/kdfs/x942kdf.c [hunk 1]
```diff
@@ -165,6 +169,9 @@ static int der_encode_sharedinfo(WPACKET *pkt, unsigned char *buf, size_t buflen
  * suppPubInfo (in bits). It is ignored if the value is 0.
  * |cek_oid| The oid of the key wrapping algorithm.
  * |cek_oidlen| The length (in bytes) of the key wrapping algorithm oid,
+ * |acvp| is the optional blob of DER data representing all the fields related
+ * to partyUInfo, partyVInfo, suppPubInfo and suppPrivInfo. It can be NULL.
+ * |acvp_len| is the |acvp| length (in bytes).
```
Comment (levitte) [inline providers/implementations/kdfs/x942kdf.c:174]: I would amend this comment to have it make it clear that when acvp is non-NULL, the four *blah*Info MUST be NULL.

**Review by levitte — CHANGES_REQUESTED** [review]

File: providers/implementations/kdfs/x942kdf.c (hunk 1 above)
Comment (slontis) [reply providers/implementations/kdfs/x942kdf.c:174]: I left it out because the caller was responsible for the check.. Will add anyway.

**Comment by slontis** [comment]
please reapprove..

**Review by paulidale — APPROVED** [review]
Still good.

**Comment by slontis** [comment]
Thanks. Merged to master.

**Pull request closed without merging**